frontend/app/components/loading-button.tsx

```diff
@@ -0,0 +1,72 @@
+import type { ComponentProps } from 'react';
+import { forwardRef } from 'react';
+
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import type { FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
+
+import { Button } from './button';
+import { ButtonEndIcon, ButtonStartIcon } from './button-icons';
+
+type ButtonEndIconProps = ComponentProps<typeof ButtonEndIcon>;
+type ButtonStartIconProps = ComponentProps<typeof ButtonStartIcon>;
+
+export interface LoadingButtonProps extends ComponentProps<'button'> {
+  endIcon?: ButtonEndIconProps['icon'];
+  endIconProps?: OmitStrict<ButtonEndIconProps, 'icon'>;
+  pill?: boolean;
+  startIcon?: ButtonStartIconProps['icon'];
+  startIconProps?: OmitStrict<ButtonStartIconProps, 'icon'>;
+  loading?: boolean;
+  loadingIcon?: FontAwesomeIconProps['icon'];
+  loadingIconProps?: OmitStrict<FontAwesomeIconProps, 'icon'>;
+  loadingPosition?: 'end' | 'start';
+}
+
+const LoadingButton = forwardRef<HTMLButtonElement, LoadingButtonProps>(
```

Comment:
`forwardRef<>()` should no longer be required in React 19.

<https://react.dev/blog/2024/12/05/react-19#ref-as-a-prop>

Reply:
We will fix that in other components in another PR. Good point @gregory-j-baker .

Reply:
Why not fix it now?